### PR TITLE
Gleeful demolition

### PR DESCRIPTION
**Pull request by AhmadYProjects**

### CODE REVIEW

**Comment by AhmadYProjects** [comment]
[[Gleeful Demolition]]

File: Mage.Sets/src/mage/cards/g/GleefulDemolition.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    public GleefulDemolition copy() {
+        return new GleefulDemolition(this);
+    }
+}
+
+class GleefulDemolitionEffect extends DestroyTargetEffect {
+    GleefulDemolitionEffect(){
+
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/g/GleefulDemolition.java:50]: don't forget the super constructor here

File: Mage.Sets/src/mage/cards/g/GleefulDemolition.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        super(card);
+    }
+
+    @Override
+    public GleefulDemolition copy() {
+        return new GleefulDemolition(this);
+    }
+}
+
+class GleefulDemolitionEffect extends DestroyTargetEffect {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/g/GleefulDemolition.java:48]: this doesn't really need to extend `DestroyTargetEffect`, it can just be a `OneShotEffect`

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/g/GleefulDemolition.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                permanent == null || player == null) {
+            return false;
+        }
+        boolean isMine = permanent.isControlledBy(source.getControllerId());
+        if (isMine) {
+            Token token = new PhyrexianGoblinToken();
+            token.putOntoBattlefield(3,game,source);
+
+        }
+        return super.apply(game,source);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/g/GleefulDemolition.java:79]: this super call doesn't do anything anymore, replace it with `return true` and add `permanent.destroy` between lines 73 and 74

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Pull request merged**